### PR TITLE
Add converters between API types and Endpoint types

### PR DESCRIPTION
**Pull request by inlined**
Add alternate converters functionToEndpoint and endpointToFunction based on functionToSpec and specToFunction. I highly recommend viewing this by commit. The first commit is a mechanical copy + paste + function name change and the second commit is actually interesting.

### CODE REVIEW

File: src/gcp/cloudfunctions.ts [hunk 1]
```diff
@@ -579,44 +579,47 @@ export function functionFromSpec(
...
+  let trigger: backend.Triggered;
   let uri: string | undefined;
   if (gcfFunction.httpsTrigger) {
+    trigger = { httpsTrigger: {} };
+    uri = gcfFunction.httpsTrigger.url;
+  } else if (gcfFunction.labels?.["deployment-scheduled"]) {
     trigger = {
-      // Note: default (empty) value intentionally means true
-      allowInsecure: gcfFunction.httpsTrigger.securityLevel !== "SECURE_ALWAYS",
+      scheduleTrigger: {},
```
Comment (taeold) [inline src/gcp/cloudfunctions.ts:591]: q: why do we not copy over attributes like `schedule`, `timeZone`, and `retryConfig` here?

File: src/gcp/cloudfunctionsv2.ts [hunk 2]
```diff
@@ -510,86 +511,80 @@ export function functionFromEndpoint(cloudFunction: backend.FunctionSpec, source
...
       for (const { attribute, value } of gcfFunction.eventTrigger.eventFilters || []) {
-        trigger.eventFilters[attribute] = value;
+        trigger.eventTrigger.eventFilters[attribute] = value;
       }
     }
   } else {
-    trigger = {
-      allowInsecure: false,
-    };
+    trigger = { httpsTrigger: {} };
```
Comment (taeold) [inline src/gcp/cloudfunctionsv2.ts:569]: q: what about `uri`?

tangent: does GCFv2 expose url for all functions types (event, http) like Cloud Run?

**Review by taeold — APPROVED** [review]

File: src/gcp/cloudfunctions.ts (hunk 1 above)
Comment (inlined) [reply src/gcp/cloudfunctions.ts:591]: We would have to make a second API request to Cloud Scheduler to fill in these details and historically we have not done this.

File: src/gcp/cloudfunctionsv2.ts (hunk 2 above)
Comment (inlined) [reply src/gcp/cloudfunctionsv2.ts:569]: URI is an attribute of Function, not httpsTrigger, and is set a few lines below (see line 584). It is present for all trigger types.

**Pull request merged**